### ISSUE TITLE
linter: no-unused-vars shouldn't raise on ambient namespace declarations

### ISSUE THREAD

**Issue by mary-ext** [oxc-project/oxc#17122]
### What version of Oxlint are you using?

1.34.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

no custom config used

### What happened?

```
export namespace Foo {
  type foo = 123;
//     ^^^ this is correctly flagged
}

export declare namespace Foo {
  type foo = 123;
//     ^^^ this should not be flagged, as it is exported
}
```

taken from https://github.com/mary-ext/atcute/blob/6baf550495fbca37e3c3a053a6f710d8c68fc569/packages/lexicons/lexicons/lib/types/brand.ts#L1-L11

**Comment by TheAlexLichter** [comment]
(Simple [REPL link](https://playground.oxc.rs/#eNqdVE1rGzEQ/StCZ5cmKYXg0kNJaiikTdqU5mICWml2rUQrbaXZOMb4v/dpP2QfcqoxWHojzbx58+S91HIp6bULkYVXLaVOaRKrEMR+7YXgXUeixu6zOL/48Gnt378X+fP4+Ch4Y5PAV4cYSbPbidqppiGz9oe1X/spqyHtVKT/yp42oXdG+MCiwskx/UIo1OVceiwxVJQLGeRyL2Pv84+znuWSY08LWYfYKmaKclkrl4BwVD5luCA2BaeQ6Xpgq9gGn0pwu7E8ci9Qq3zjjlsd2i5SOl5JOnQIjwTSrq2Cm3e6bqZjh4XsVEyZ2B5DYPIJdTEQTq/oRzkXtr+I++hve07W0Kr3OlObU+WaFF/oDgJnviM63Ptz+c1ztMioSyARaLPVX2MMcULBIWs1cMD6RKu97BP9VtWxKVbVgzW8kcuLhSRvbusb69GldDXodqjGU/zyDNVsluhnH/io01N6vX8D/pt3dzF0qCVVeueJDBnkxKAs6DVXoW1VjjkHFH3Y0ix62c79I75Vu4QjVVT6mfgeQ8P1+fCMwooj86l+qJ7g4IeoOuSYRc2FBq5fGEJWPRhSPLkGtYqPRr1YxYZgO0np4uz8IxJAwWuqcWcV4pVTKa0sOVMGAv9StC15Vg7OCzBeGQyireWCfidWRjFEmKb2tmP3MoFsh9lT9Aqem6nqYKih4W34PONix6dkAv4EJjU93tqQa0YcNfPZIcvs82yW+RGg89Dd0AuVpM9E3Y/84E8JwI/BreDNfOOFYhXSiZbW5xnkUFkdAJtBvgyX1SHjWea5p8M/WTWw+w==), correct behavior in [`ts-eslint` playground](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=KYDwDg9gTgLgBAOwIYFtgGcxIMbDgMQgjgG8AoOOGATzDwDMi4BeOARgCYBmAbgrgD0AyiIB64qgAsAlujiy42aFGDYYAG2px66pAHM9wACZkAvmTKhIsOEdW6ViVBiy4CTcpRp1tTVp15%2BIRFKcVEpBXRJCABXdSNECHgAIwZdA2MAGjgkOWl4BStoGGMzIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))

**Comment by mary-ext** [comment]
semi-related ([REPL](https://playground.oxc.rs/#eNptVF1r2zAU/StCTxtksBUGxfuiaxcYtGvXlvVh2YNsXbtqZcmTrpOUkP++I9tyCisEIp1zP8+98k5WspCaKqsCidbr3pJYybVhirySYrdyQhjHFGpVkTiJkQIb7z7eik9CuafPo4UQ7L/ShdcUVKKvKfaWX42UELTtqGLSXwpxsLlFhpseiUbjpVXN7z+L7FJ5JN0yPCIH45qZGK+mNqRHx/9NTON8oFPVRwJZem9JuYl9XYi1N/pDuu1XDj+5kF4WOxl6l/4supUFh54WsvahVYzmM8BBuZhQWdTKRiAmeqvQ2tkg4dBYnMnNfdKxg3Iz1CrX2MO18m0XKB5cYuU70GO6+NSW3uZbVTeT2X4hOxUwilQwVCIXkReD5LhFO8pav7km7oO77DkaTcveVam0HCrlpLCmK0w91Tuig9+v4+8uyRlNNRORUDab6lsIPkwoakhSDTXg/EyqnYTwt6o8NMWqvDOa72VxtJDk9GV9bhy6lLZGuR2y8cQfv0U2zNLSz97zQaeHuL15Af6bblfBd8glVXzjiDRpxMSgDMprTn3bqsRZCxR9mLlZ9LLJ/YPfqKcIkzKo6pH4BkODezbOqGpprHzK78sHLPZdUB1iZFFToqHWE4aQZY8KKTxzg1rzHo16sQoNYeskxaO3794jABQ8oxo+Sx9OrYpxacjqeSDpQQXTkmNlsXkeizcPBmxreEYviJVWDBGmqb28sTuZHlGH2VNwCjuXS63wYBsangZeVZuoMctD1B4fj0lN5zwPsTJiqcm2Q5S852lZ8iNA5747pzXNQR+Juh9QOe/O4Ip99HaJ3Uweawqlj8+0NC7NIFHzaQ9YD/IleD7tE55kPizr/h85n8Ul))

```ts
declare module "vitest" {
  interface Assertion<T = any> {
//                    ^ it should not flag this generic as unused
    toBeModerationResult(
      expected?: ModerationTestSuiteResultFlag[],
      context?: string,
      stringifiedResult?: string,
      ignoreCause?: boolean,
    ): void;
  }
}
```

TypeScript will raise `All declarations of 'Assertion' must have identical type` if prefixed or renamed

**Comment by camc314** [comment]
> TypeScript will raise All declarations of 'Assertion' must have identical type if prefixed or renamed

it's not possible for us to analyze this statically without type information to avoid reporting this false positive

**Comment by mary-ext** [comment]
> it's not possible for us to analyze this statically without type information to avoid reporting this false positive

ah I think the linter should just not flag unused generics in this case considering that any type generics declared would be because someone is trying to extend the interface of another module, no type analysis required
